### ISSUE TITLE
9.12 -> 10 migration error: sentry.models.group.MultipleObjectsReturned: get() returned more than one Group -- it returned 2!

### ISSUE THREAD

**Issue by Fogapod** [getsentry/onpremise#341]
I had problem with this migration before, it was fixed yesterday: https://github.com/getsentry/onpremise/issues/322, but there is a new one now. Log:
```python
Applying sentry.0024_auto_20191230_2052.../usr/local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:1430: RuntimeWarning: DateTimeField Event.datetime received a naive datetime (2019-10-12 14:23:18.998532) while time zone support is active.
  RuntimeWarning)
/usr/local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:1430: RuntimeWarning: DateTimeField Event.datetime received a naive datetime (2020-01-10 14:23:18.998532) while time zone support is active.
  RuntimeWarning)
Events to process: 800

Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 164, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 30, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/commands/upgrade.py", line 168, in upgrade
    _upgrade(not noinput, traceback, verbosity, not no_repair)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/commands/upgrade.py", line 130, in _upgrade
    ignore_ghost_migrations=True,
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 130, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python2.7/site-packages/sentry/new_migrations/monkey/executor.py", line 26, in apply_migration
    state, migration, fake=fake, fake_initial=fake_initial
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/operations/special.py", line 189, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/usr/local/lib/python2.7/site-packages/sentry/migrations/0024_auto_20191230_2052.py", line 71, in backfill_eventstream
    if event.project is None or event.group is None:
  File "/usr/local/lib/python2.7/site-packages/sentry/models/event.py", line 76, in group
    self._group_cache = Group.objects.get(id=self.group_id)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 389, in get
    (self.model._meta.object_name, num)
sentry.models.group.MultipleObjectsReturned: get() returned more than one Group -- it returned 2!
```
and
```
14:23:20 [WARNING] sentry.eventstream.kafka.backend: Could not publish message (error: KafkaError{code=MSG_SIZE_TOO_LARGE,val=10,str="Broker: Message size too large"}): <cimpl.Message object at 0x7fb7e2947950>
```
(repeated 20 times)

After fix I was able to skip this migration with `SENTRY_SKIP_EVENTS_BACKFILL_FOR_10=1`, but I would want to keep my events.

Side note: placing `SENTRY_SKIP_EVENTS_BACKFILL_FOR_10=1` into `.env` did not work, I had to put it into docker-compose file, is is a bug?

**Comment by BYK** [comment]
>Side note: placing SENTRY_SKIP_EVENTS_BACKFILL_FOR_10=1 into .env did not work, I had to put it into docker-compose file, is is a bug?

This is expected as you'd need to explicitly pass it through the docker-compose file.

Investigating the other issues.

**Comment by BYK** [comment]
>sentry.models.group.MultipleObjectsReturned: get() returned more than one Group -- it returned 2!

This one seems to be related to your manual DB changes you mentioned earlier. We cannot fix or protect against this as it will cause other issues everytime you try to view those events with more than one group associated. You should simply fix your database for this. The ORM should not allow this to happen in the first place.

>14:23:20 [WARNING] sentry.eventstream.kafka.backend: Could not publish message (error: KafkaError{code=MSG_SIZE_TOO_LARGE,val=10,str="Broker: Message size too large"}): <cimpl.Message object at 0x7fb7e2947950>

This one means we need to increase `max.messages.bytes` for Kafka. Our Kafka expert dug up this: https://stackoverflow.com/a/21343878/90297

I'll be looking into this some time this week.

**Comment by shaula** [comment]
> After fix I was able to skip this migration with `SENTRY_SKIP_EVENTS_BACKFILL_FOR_10=1`, but I would want to keep my events.

I was experiencing the very same issue, but was able to finish migration without this env SENTRY_SKIP_EVENTS_BACKFILL_FOR_10 by doing:
```
git checkout c95eb56
SENTRY_IMAGE=getsentry/sentry:b0ea6ea032d886edf24a70c1d1ac0d79e1462b39 ./install.sh
git checkout master
./install.sh
docker-compose up -d --force-recreate
```

**Comment by BYK** [comment]
This should be fixed by #497.

**Comment by SebastianZ84** [comment]
Hi guys,
i have the same error during the update as described above.
Can someone explain to me how to find the messages that are in more than one group?
